### ISSUE TITLE
support Taiga (or any other issue tracker) as a human approval step in a Jenkins CD pipeline

### ISSUE THREAD

**Issue by jstrachan** [fabric8io/fabric8#4087]
so that a build step raises an issue with a link to the build / version / staging environment etc in the issue text.

Then if the issue is closed ; marked as fixed; then the next step in the pipeline is triggered.

Folks can then use a project in Taiga for a team of reviewers etc. 

The other option is to try use gerrit code reviews to approve a git commit change which represents which version number(s) of apps should be deployed to an environment. Then a build on a git commit would trigger the deployment?

**Comment by christian-posta** [comment]
Taiga has webhooks: http://taigaio.github.io/taiga-doc/dist/webhooks.html
Not sure how it should kick jenkins though... maybe a very simple jenkins plugin that exposes an endpoint for a taiga web hook? or fabric8 hosts the endpoint and can interact w/ jenkins? https://wiki.jenkins-ci.org/display/JENKINS/Remote+access+API 

not sure yet :)

**Comment by jstrachan** [comment]
ah cool thanks :)

I've raised a separate issue for gerrit approvals: https://github.com/fabric8io/fabric8/issues/4088 which sounds like the simplest thing to implement really ;)

**Comment by rawlingsj** [comment]
Yeah we already have taiga working with the lets chat and gogs webhooks, so tasks and issues are automatically moved, closed etc using commit messages. It works really nicely.

I agree on having a webhook configured with fabric8 and I was thinking hubot would be great for this.  We could then integrate custom usecases  for Jenkins, Jira, Gerrit, anything that has an api.

**Comment by rawlingsj** [comment]
Also by doing this we can tie in chat into approvals. Hubot could notify a room / individual requesting approval for a deployment etc.
